pkg/app/piped/livestatestore/cloudrun/store.go

```diff
@@ -30,10 +32,57 @@ type store struct {
 	client provider.Client
 }
 
+type app struct {
+	service provider.ServiceManifest
+	// The states of service and all its active revsions which may handle the traffic.
+	states  []*model.CloudRunResourceState
+	version model.ApplicationLiveStateVersion
+}
+
 func (s *store) run(ctx context.Context) error {
+	svcs, err := s.getAllServices(ctx)
+	if err != nil {
+		return fmt.Errorf("failed to get all cloudrun services: %w", err)
+	}
+	revs := s.listRevisionsFromServices(ctx, svcs)
+
+	// Update apps to the latest.
+	s.storeApps(svcs, revs)
+
+	return nil
+}
+
+func (s *store) storeApps(svcs []*provider.Service, revs map[string][]*provider.Revision) {
+	apps := make(map[string]app, len(svcs))
+	for i := range svcs {
+		sm, err := svcs[i].ServiceManifest()
+		if err != nil {
+			s.logger.Error("failed to load cloudrun service into service manifest", zap.Error(err))
+			continue
+		}
+
+		appID, ok := sm.AppID()
+		if !ok {
+			continue
+		}
+
+		id, _ := svcs[i].UID()
+		now := time.Now()
+		apps[appID] = app{
+			service: sm,
+			states:  provider.MakeResourceStates(svcs[i], revs[id], now),
+			version: model.ApplicationLiveStateVersion{
+				Timestamp: now.Unix(),
+			},
+		}
+	}
```

Comment:
nit: Maybe we just need this part in a function `buildAppMap` so it can be testable.

`s.apps.Store(apps` could be moved to inside `run`